robot_controllers/include/robot_controllers/diff_drive_base.h

```diff
@@ -169,15 +169,16 @@ class DiffDriveBaseController : public Controller
   ros::Time last_update_;
   ros::Duration timeout_;
 
+  // The odom mutex protects the following variables
   boost::mutex odom_mutex_;
```

Comment:
we should rename this mutex to "msg_mutex_" or similar -- since it no longer covers just odom_ -- it appears to only be used in 2 places, so the update is not too bad.